### DISCUSSION TITLE
React Widget not working

### DISCUSSION THREAD

**Discussion by vibsid0986** [eclipse-theia/theia#13707] [Q&A]
Hello Team,

We have created a React application which provides the functionality to drag and drop components onto canvas. Please refer below Screenshot for your reference.
![image](https://github.com/eclipse-theia/theia/assets/98730468/6f677372-cff6-4594-8f04-8e5a707fc5cd)

We are facing an issue as post integrating this application as react widget, Drag feature is working as expected, however elements are not dropping onto canvas. It seems like post integration with theia application, dropTarget is unrecognisable within application, it cannot understand if the element is being hovered over the canvas. Request you to help us understand what could be the reason for same.

**Comment by planger** [comment]
Hello,

Thanks for reaching out. It's a bit challenging to diagnose the problem without seeing and debugging your code directly. Integration issues like these can be tricky!

However, integrating functionality similar to what you're describing is indeed possible. We've implemented something like this in the [Theia integration of the GLSP project](https://github.com/eclipse-glsp/glsp-theia-integration) for diagram editors. Although it's not a React widget, you might find some insights by looking at the implementation of the [GLSP Diagram Widget](https://github.com/eclipse-glsp/glsp-theia-integration/blob/master/packages/theia-integration/src/browser/diagram/glsp-diagram-widget.ts).

For the use cases where we've implemented drag and drop (and not "click palette, click canvas"), I don't remember anything specific we had to do to make it work. Maybe you are dealing with event propagation or Z-index issues?

**Comment by vibsid0986** [comment]
Hello @planger ,

Thanks for your prompt response. 

Just to give you a more clear picture, we are using react-dnd library for drag and drop functionality and not utilizing GLSP Diagram Widget. React DND uses HTML5 drag and drop API under the hood and hides its quirks. 

This library automatically triggers hover event when any component is dragged over canvas and provide isOver state set to true, however this is not happening when we integrated it as react widget in theia.

We would like to understand if Theia supports HTML5 drag and drop API or is there any other way to achieve the same task. 
Please find below link for react-dnd for your reference.

[https://react-dnd.github.io/react-dnd/about](url)

Request you to provide assistance on same.

Thanks in advance.

**Reply by planger** [reply]
Oh, ok! I'm afraid I haven't used `react-dnd` in Theia yet. So I'm not sure if there is some specialty or any other reason why this wouldn't just work. I'm sorry!
